test/utils/deploy/hypershift.go

```diff
@@ -91,6 +95,25 @@ func (h *HyperShiftNRO) Teardown(ctx context.Context, timeout time.Duration) {
 		Expect(wait.ForConfigToBeReady(ctx, e2eclient.MNGClient, np.Name, np.Namespace)).To(Succeed())
 
 		Expect(e2eclient.MNGClient.Delete(ctx, h.KcConfigMapObj)).To(Succeed())
+
+		By("checking that generated configmap has been deleted")
```

Comment:
IMO this is not the ideal place for checking the deletion. 
It would be more appropriate in the `uninstall` suite.
But after checking the `uninstall` suite I see it completely redundant.
We're deleting the nro and kubeletconfig in the `uninstall` suite but these objects were already removed here:
https://github.com/openshift-kni/numaresources-operator/blob/7de8e4b52dcff68106fa0efa8f5b218acfdf423a/test/e2e/install/install_test.go#L203 

In the nutshell the `uninstall` need to go through some refactoring but this is another issue for another PR.

Reply:
make sense to me with the catch that suites should not depend on each other as much as possible; IOW they should be indpedent and self-sufficient.